src/lm_buddy/integrations/vllm.py

```diff
@@ -12,8 +12,26 @@ class InferenceServerConfig(BaseLMBuddyConfig):
 
     Note: This configuration is intended to be generic and not bound to the interface
     of any specific training/evaluation framework. See `LocalChatCompletionConfig`
-    for intended usage alongside a third-party framework.
+    or `vLLMCompleptionsConfig` for intended usage alongside a third-party framework.
     """
 
     base_url: str
     engine: str | HuggingFaceAssetPath | None = None
+
+
+class VLLMCompletionsConfig(BaseLMBuddyConfig):
+    """Configuration for a vLLM-based completions service
+
+    The "local-chat-completions" model is powered by a self-hosted inference server,
+    specified as an `InferenceServerConfig`. Additional arguments are also provided
+    to control the tokenizer type and generation parameters.
+    """
+
+    inference: InferenceServerConfig
+
+    # vLLM-specific params
```

Comment:
do these come from here? we might want to add a note in case we decide to add more later https://github.com/vllm-project/vllm/blob/main/vllm/sampling_params.py

Reply:
Yes, this is why I wanted to have a vLLM-specific configuration, as the parameters can be way more than those which are prometheus-specific.
Also, if we ever decide to support other types of inference service (e.g. TGI which is what the original Prometheus was built on) which are not API-compatible we can just have a new config for them without having to tamper with the prometheus-specific params.
TL/DR: yes, I will add the URL for reference, that's a great idea!